dvc/command/ls/__init__.py

```diff
@@ -34,7 +34,11 @@ def run(self):
                 recursive=self.args.recursive,
                 dvc_only=self.args.dvc_only,
             )
-            if entries:
+            if self.args.show_json:
+                import json
+
+                logger.info(json.dumps(entries))
```

Comment:
[Q] What is the format of this output? 

Is it:
```
[{"path1": "fileOrDirectory"}, {"path2": "fileOrDirectory"}, {"path3": "fileOrDirectory"}]
```
?

Reply:
@mattseddon 
Here is output for `dvc list --dvc-only git@github.com:iterative/dvc.git scripts/innosetup --show-json`
```bash
[{"isout": true, "isdir": false, "isexec": false, "path": "dvc.ico"}, {"isout": true, "isdir": false, "isexec": false, "path": "dvc_left.bmp"}, {"isout": true, "isdir": false, "isexec": false, "path": "dvc_up.bmp"}]
```

Reply:
Thank you

Reply:
@pared, maybe it'd be better to have the schema as an object rather than a list?

Reply:
@skshetry I am not sure I follow - do you mean to have a schema and validate the resulting list or return the results as some kind of an object?

Reply:
@pared, Sorry, I meant the result that we return. List might not be that useful, I think. Either a dictionary (or, a hierarchy inside a dictionary) might be easier?

Reply:
@skshetry it might be that dict would be better, we could easily preserve dir structure, especially for (-R). @mattseddon what do you think about that? Would that be desirable feature for you?

Reply:
I'm quite happy with what we have at the moment but I can deal with you updating to whatever you see fit. 

However, any dict with a 'hierarchy' would be undesirable as would require more parsing after we've run the command.